source/json.c

```diff
@@ -0,0 +1,306 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/string.h>
+
+#include <aws/common/json.h>
+
+#include <aws/common/external/cJSON.h>
+
+static struct aws_allocator *s_aws_json_module_allocator = NULL;
+static bool s_aws_json_module_initialized = false;
+
+struct aws_json_value *aws_json_new_string(struct aws_allocator *allocator, const struct aws_byte_cursor cursor) {
+    struct aws_string *tmp = aws_string_new_from_cursor((struct aws_allocator *)allocator, &cursor);
+    void *ret_val = cJSON_CreateString(aws_string_c_str(tmp));
+    aws_string_destroy_secure(tmp);
+    return ret_val;
+}
+
+struct aws_json_value *aws_json_new_number(struct aws_allocator *allocator, double number) {
+    (void)(allocator); // prevent warnings over unused parameter
+    return (void *)(uintptr_t)cJSON_CreateNumber(number);
```

Comment:
I'm still confused why uintptr_t is used here.

Reply:
It is to prevent a compiler warning on Windows that otherwise makes the CI fail, if I recall correctly. I'll see if I can remove it since the code has changed a bit since it was added. 👍